### PR TITLE
docs: added a setting status page.

### PR DESCRIPTION
**Pull request by Jaskowicz1**
This PR adds a new page for setting the bot's status.

- [x] My pull request is made against the `dev` branch.
- [x] I have ensured that the changed library can be built on your target system. I did not introduce any platform-specific code.
- [x] I have ensured that all methods and functions are **fully documented** using doxygen style comments.
- [x] I tested my commits, by adding a test case to the unit tests if needed
- [x] I have ensured that I did not break any existing API calls.
- [x] My code follows the [coding style guide](https://dpp.dev/coding-standards.html) (if you are not sure, match the code style of existing files including indent style etc).
- [x] I have not built my pull request using AI, a static analysis tool or similar without any human oversight. Where I have generated this pull request using a tool, I have justified why this is needed.

### CODE REVIEW

**Comment by Jaskowicz1** [comment]
Corrected the misc.md subpage creation, removing the `"<title>"` from the line.

**Comment by Henonicks** [comment]
Too much cod, but it's beatiful!11!!1!1!11

**Review by braindigitalis — CHANGES_REQUESTED** [review]
this would be better if we then show in a second example how to set the status on a timer e.g. with a guild count in it. pay attention though to the fact that the presence is set per shard so you'll need to loop all shards to set presence properly

**Comment by Jaskowicz1** [comment]
> this would be better if we then show in a second example how to set the status on a timer e.g. with a guild count in it. pay attention though to the fact that the presence is set per shard so you'll need to loop all shards to set presence properly

I like that idea! I'll get onto that now.

**Comment by Jaskowicz1** [comment]
> this would be better if we then show in a second example how to set the status on a timer e.g. with a guild count in it.

The suggested example has now been added and is ready for a review!

File: docpages/example_programs/misc/setting_status.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+{
+    /* Create the bot */
+    dpp::cluster bot("token");
+
+    bot.on_log(dpp::utility::cout_logger());
+
+    bot.on_ready([&bot](const dpp::ready_t& event) {
+        /* We don't need the run_once here as we're not registering commands! */
+
+        std::thread status_thread([&bot]() {
```
Comment (braindigitalis) [inline docpages/example_programs/misc/setting_status.md:49]: the thread needs to be in a run_once or the bot will spawn a new thread every time any shard reconnects

Comment (Jaskowicz1) [reply docpages/example_programs/misc/setting_status.md:49]: Ah yes, good point.

**Review by braindigitalis — APPROVED** [review]

**Pull request merged**